### PR TITLE
Update `TestCase::assertLocalizedHolidays`: add subdivisions holidays support

### LINKED ISSUE #2115: Enhance `TestCase::assertLocalizedHolidays`
`TestCase::assertLocalizedHolidays` now considers all categories of holidays, but doesn't consider subdivisions holidays.  However, in Switzerland, for example, there are only a few national holidays, and the test looks like this:
```python
    def test_l10n_en_us(self):
        self.assertLocalizedHolidays(
            "en_US",
            ("2023-01-01", "New Year's Day"),
            ("2023-05-18", "Ascension Day"),
            ("2023-08-01", "National Day"),
            ("2023-12-25", "Christmas Day"),
        )
```
If we include provincial holidays, the test will look like this:
```python
    def test_l10n_en_us(self):
        self.assertLocalizedHolidays(
            "en_US",
            ("2023-01-01", "New Year's Day"),
            ("2023-01-02", "Berchtold's Day"),
            ("2023-01-06", "Epiphany"),
            ("2023-03-01", "Republic Day"),
            ("2023-03-19", "Saint Joseph's Day"),
            ("2023-04-07", "Good Friday"),
            ("2023-04-10", "Easter Monday"),
            ("2023-04-13", "Battle of Naefels Victory Day"),
            ("2023-05-01", "Labor Day"),
            ("2023-05-18", "Ascension Day"),
            ("2023-05-29", "Whit Monday"),
            ("2023-06-08", "Corpus Christi"),
            ("2023-06-23", "Independence Day"),
            ("2023-06-29", "Saints Peter and Paul"),
            ("2023-08-01", "National Day"),
            ("2023-08-15", "Assumption Day"),
            ("2023-09-07", "Genevan Fast"),
            ("2023-09-18", "Prayer Monday"),
            ("2023-09-25", "Saint Nicholas of Flüe"),
            ("2023-11-01", "All Saints' Day"),
            ("2023-12-08", "Immaculate Conception"),
            ("2023-12-25", "Christmas Day"),
            ("2023-12-26", "Saint Stephen's Day"),
            ("2023-12-31", "Restoration Day"),
        )
```
Should we do this or is it an impractical overkill?

### PR DESCRIPTION
**Pull request by KJhellico**
## Proposed change

Update `TestCase::assertLocalizedHolidays`: add subdivisions holidays support.
Resolve #2115.

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new `holidays` functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've successfully run `make check`, all checks and tests are green

[contributing-guidelines]: https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/holidays/tree/dev/docs/source

### CODE REVIEW

**Review by PPsyrius — APPROVED** [review]
LGTM 🌐

File: tests/common.py [hunk 1]
```diff
@@ -306,7 +306,17 @@ def _assertLocalizedHolidays(self, localized_holidays, language=None):  # noqa:
         )
 
         # Populate holidays for an entire year.
-        self.assertIn(localized_holidays[0][0], instance)
+        _ = localized_holidays[0][0] in instance
```
Comment (arkid15r) [inline tests/common.py:309]: Consider removing this line and initializing instance using

```
instance = self.test_class(
    years=int(localized_holidays[0][0].split("-")[0]),
    language=language,
    categories=self.test_class.supported_categories,
)
```

File: tests/common.py [hunk 2]
```diff
@@ -306,7 +306,17 @@ def _assertLocalizedHolidays(self, localized_holidays, language=None):  # noqa:
...
+
+        for subdiv in self.test_class.subdivisions:
+            subdiv_instance = self.test_class(
+                subdiv=subdiv,
+                years=instance.years,
+                language=language,
+                categories=self.test_class.supported_categories,
+            )
+            for dt, name in subdiv_instance.items():
+                instance[dt] = name
```
Comment (arkid15r) [inline tests/common.py:319]:
```suggestion
        for subdiv in instance.subdivisions:
            instance.update(
                self.test_class(
                    subdiv=subdiv,
                    years=instance.years,
                    language=language,
                    categories=instance.supported_categories,
                )
            )
```

**Review by arkid15r — COMMENTED** [review]
LGTM

**Review by arkid15r — APPROVED** [review]

**Pull request merged**